app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java

```diff
@@ -0,0 +1,142 @@
+package edu.grinnell.grinnell_publications_android.Fragments;
+
+
+import android.graphics.PorterDuff;
+import android.graphics.drawable.Drawable;
+import android.support.annotation.Nullable;
+import android.os.Bundle;
+
+import android.support.design.widget.CollapsingToolbarLayout;
+import android.support.design.widget.FloatingActionButton;
+import android.support.design.widget.Snackbar;
+import android.support.v4.app.Fragment;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.Toolbar;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import edu.grinnell.grinnell_publications_android.Models.Interfaces.UserInterface;
+import edu.grinnell.grinnell_publications_android.R;
+
+import static android.support.design.widget.Snackbar.LENGTH_SHORT;
+
+
+/**
+ * Fragment displays articles and allows users to mark article as favorited.
+ * @author Yazan Kittaneh
+ */
+
+public class ExpandedArticleFragment extends Fragment implements UserInterface {
+
+    private ImageView mHeaderImage;
+    private CollapsingToolbarLayout mCollapsingToolbar;
+    private FloatingActionButton mFavoriteButton;
+    private TextView mArticleContent;
+    private Toolbar mArticleToolbar;
+
+    private boolean isFavorited = false;
+    private int colorFilter;
+    private String snackBarMessage;
+
+    public ExpandedArticleFragment() {}
+
+    public static ExpandedArticleFragment newInstance() {
+        return new ExpandedArticleFragment();
+    }
+
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater,
+                             ViewGroup container,
+                             Bundle savedInstanceState) {
+        final View expandedArticleFragment = inflater.inflate(R.layout.fragment_extended_article, container, false);
+        setViewBindings(expandedArticleFragment);
+        initializeUI();
+        return expandedArticleFragment;
+    }
+
+    @Override
+    public void initializeUI(){
+        mArticleToolbar.setNavigationIcon(ContextCompat.getDrawable(getContext(), R.drawable.ic_action_back));
+        loadPlaceHolderData();
+        setOnClickListeners();
+    }
+
+    public void setViewBindings(View v){
```

Comment:
This method should not be public. It is called within this class only so maintain it as private. Another overhead is that if you make this code public, I can assume that a client calling this method from another class can pass in a null. Your code will throw an NPE if that should happen. Should you make it private, a client using it in this class can see the implementation and knows for sure that passing in null might lead to an NPE. To remove all of this overhead, you should always be defensive and not assume your clients know what the arguments will be. (i.e check for null inputs, one means of being defensive) If you want to maintain some level of assumption, please write pre and post condition statements.